samples/bluetooth/broadcast_audio_source/src/main.c

```diff
@@ -68,17 +138,81 @@ static K_SEM_DEFINE(sem_stopped, 0U, ARRAY_SIZE(streams));
 #define BROADCAST_SOURCE_LIFETIME  120U /* seconds */
 
 #if defined(CONFIG_LIBLC3)
-
-#include "lc3.h"
-
-static lc3_encoder_t lc3_encoder;
-static lc3_encoder_mem_16k_t lc3_encoder_mem;
 static int freq_hz;
 static int frame_duration_us;
 static int frames_per_sdu;
 static int octets_per_frame;
 
-static void init_lc3(void)
+static K_SEM_DEFINE(lc3_encoder_sem, 0U, ARRAY_SIZE(streams));
+#endif
+
+static void send_data(struct broadcast_source_stream *source_stream)
+{
+	struct bt_bap_stream *stream = &source_stream->stream;
+	struct net_buf *buf;
+	int ret;
+
+	if (stopping) {
+		return;
+	}
+
+	buf = net_buf_alloc(&tx_pool, K_FOREVER);
+	if (buf == NULL) {
+		printk("Could not allocate buffer when sending on %p\n",
+		       stream);
+		return;
+	}
+
+	net_buf_reserve(buf, BT_ISO_CHAN_SEND_RESERVE);
+#if defined(CONFIG_LIBLC3)
+	uint8_t lc3_encoded_buffer[preset_active.qos.sdu];
```

Comment:
I would leave this for when/if it would be a problem to keep the code more understandable at this point.  Unless you want to always just want the max size?